eureka/src/main/java/com/linecorp/armeria/server/eureka/EurekaUpdatingListener.java

```diff
@@ -328,26 +330,37 @@ public void run() {
             client.sendHeartBeat(appName, instanceId, instanceInfo, null)
                   .aggregate()
                   .handle((res, cause) -> {
-                      try (HttpData content = res.content()) {
-                          if (closed) {
-                              return null;
-                          }
+                      if (closed) {
+                          return null;
+                      }
 
-                          // The information of this instance is removed from the registry when the heart beats
-                          // fail consecutive three times, so we don't retry.
-                          // See https://github.com/Netflix/eureka/wiki/Understanding-eureka-client-server-communication#renew
-                          if (cause != null) {
-                              logger.warn("Failed to send a heart beat to Eureka: {}", client.uri(), cause);
-                          } else if (res.headers().status() != HttpStatus.OK) {
+                      if (cause != null) {
+                          logger.warn("Failed to send a heart beat to Eureka: {}", client.uri(), cause);
+                      } else {
+                          final HttpStatus status = res.status();
+
+                          if (status == HttpStatus.OK) {
+                              logger.debug("Sent a heart beat to Eureka: {}", client.uri());
+                          } else if (status == HttpStatus.NOT_FOUND) {
+                              // The information of this instance is removed from the registry when
+                              // the heart beats fail consecutive three times, so we try to re-registration.
+                              // See https://github.com/Netflix/eureka/wiki/Understanding-eureka-client-server-communication#renew
+                              logger.warn("Instance {}/{} no longer registered with Eureka." +
+                                      " Attempting re-registration.",
+                                  appName, instanceId);
```

Comment:
Could you check the indentation of this class again, please?

Reply:
Fixed